### ISSUE TITLE
Story post backend

### ISSUE THREAD

**Issue by xekc** [Lokiedu/libertysoil-site#903]
To create a story post, set `type` to `'story'` and `text_type` to either `'markdown'` or `'html'`. Next, back-end processes `text_source` according to `text_type` and outputs the result into `text`. Keep in mind that some controllers that return a collection of posts do not include `text` and `text_source` by default anymore. Add `?fields=+text,text_source` query to include those fields (prefix with `+` to extend default list of fields).

The PR also contains a complete refactoring of `createPost` and `updatePost` controllers. The motivation for this change is that it's getting harder to manage and validate all text post invariants (short_text, long_text, story). Each type of a post requires a different treatment and it makes the `createPost` and `updatePost` controllers too convoluted.

I'm also working on validation of posts using Joi. The PR may be merged as it is, I going to make a separate PR later, but for now validation errors are being sent with the `500` code instead of `400`.

* [x] Make `title` optional.
* [x] Save first image url to `more.image.url`
* [x] Save first two paragraphs of text to `more.shortText`
* [x] Exclude `text` and `text_source` columns from post lists. (all GET `/posts/*` methods)
* [x] Columns: `text_source`, `text_type`, `text`
  * [x] For non-story posts, save text into both `text_source` and `text`
  * [x] For story posts, `text` contains processed text which is safe to be rendered using `dangerouslySetInnerHTML`
  * [x] `text_type` can only be `markdown` or `html`. No plain text.
* [ ] Fix controllers after #868 is merged.

The client side needs to be refactored to use `more.shortText` instead of `text`.

**Comment by xekc** [comment]
➤ Dmitry Vdovin commented:

Travis is failing because of the linting error in the stable branch. Here is a fix: https://github.com/Lokiedu/libertysoil-site/pull/822/commits/e2f4f5d9ec244f3ef1c520c3c6c0d10db1f2b26f

Can we also stop coveralls from spamming into our PRs?

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://:/builds/12019508/badge)](https://:/builds/12019508)

Coverage increased (+0.4%) to 57.406% when pulling a22aee618a79049027997230e8886da0d475b64b on feature/extended-post-backend into 202de1b4f14cea2288f1ff02712c2a2e87a9797c on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://:/builds/12019508/badge)](https://:/builds/12019508)

Coverage increased (+0.4%) to 57.406% when pulling a22aee618a79049027997230e8886da0d475b64b on feature/extended-post-backend into 202de1b4f14cea2288f1ff02712c2a2e87a9797c on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://:/builds/12020180/badge)](https://:/builds/12020180)

Coverage increased (+0.4%) to 57.406% when pulling a22aee618a79049027997230e8886da0d475b64b on feature/extended-post-backend into 202de1b4f14cea2288f1ff02712c2a2e87a9797c on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://:/builds/12020368/badge)](https://:/builds/12020368)

Coverage decreased (-2.3%) to 54.737% when pulling a22aee618a79049027997230e8886da0d475b64b on feature/extended-post-backend into 202de1b4f14cea2288f1ff02712c2a2e87a9797c on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://:/builds/12129149/badge)](https://:/builds/12129149)

Coverage increased (+0.4%) to 57.269% when pulling 31dfadc3f3c7a1d333d50408c9bef486d1f3190a on feature/extended-post-backend into 5dd11ed75d7974336485c40ba7f20978f93eb8cc on stable.

**Comment by xekc** [comment]
➤ Vladimiras Lekecinskas commented:

@voidxnull do you think a rich (dropbox paper-like or medium.com-like) editor can be used here? What needs to be done on the back end further?

**Comment by xekc** [comment]
➤ Vladimiras Lekecinskas commented:

@indeyets could you merge this in?

**Comment by xekc** [comment]
➤ Dmitry Vdovin commented:

@xekc Absolutely. I think it's done on the back-end: now posts may be written either in Markdown or in HTML, and we already have an image loading API.

**Comment by xekc** [comment]
➤ Dmitry Vdovin commented:

Working on replacing checkit with https://github.com/hapijs/joi. The reason for this is checkit's limitations, lack of documentation, and bugs like this https://github.com/tgriesser/checkit/issues/76#issuecomment-277145908. I'm happy with the API of Joi so far.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://coveralls.io/builds/12282795/badge)](https://coveralls.io/builds/12282795)

Coverage increased (+0.5%) to 56.416% when pulling 43158e1e6104d11d59c776a663bc9f26806f6370 on feature/extended-post-backend into 153707a10539f965e76d023b3e5020ab38d0399a on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://coveralls.io/builds/12282795/badge)](https://coveralls.io/builds/12282795)

Coverage increased (+0.5%) to 56.416% when pulling 43158e1e6104d11d59c776a663bc9f26806f6370 on feature/extended-post-backend into 153707a10539f965e76d023b3e5020ab38d0399a on stable.

**Comment by xekc** [comment]
➤ Coveralls commented:

[![Coverage Status](https://coveralls.io/builds/12282917/badge)](https://coveralls.io/builds/12282917)

Coverage increased (+0.5%) to 56.416% when pulling 43158e1e6104d11d59c776a663bc9f26806f6370 on feature/extended-post-backend into 153707a10539f965e76d023b3e5020ab38d0399a on stable.
